tests/features/mocking.py

```diff
@@ -804,6 +804,34 @@ def method(cls, arg):
         flexmock(instance).should_receive("method").with_args("stuff")
         instance.method("stuff")
 
+    def test_with_args_scoped_mock(self):
+        class FooClass:
+            def method(self, arg):
+                return arg
+
+        flexmock(FooClass).should_receive("method").with_args("override").and_return("method override").once()
```

Comment:
Have you considered replacing the function altogether ? This would pass
```suggestion
        flexmock(FooClass).should_receive("method").replace_with(lambda x: "method override" if x == "override" else x).twice()
```

Reply:
Doesn't that assume we know how the inner function behaves? What if the inner function does `subprocess.call` and we want to replace it only partially.

Reply:
I see ! Yes, then I have another suggestion :
```suggestion
        flexmock(FooClass).should_call("method")
        flexmock(FooClass).should_receive("method").with_args("override").and_return("method override").once()
```
The same pattern doesn't work with `__new__` as if seems the `should_call` fails but : 
```python
        flexmock(FooClass).should_receive("__new__").replace_with(lambda _: object.__new__(FooClass))
        flexmock(FooClass).should_receive("__new__").with_args("override").and_return(fake_Foo).once()
```